osu.Game/Graphics/OsuColour.cs

```diff
@@ -63,6 +64,46 @@ public Color4 ForDifficultyRating(DifficultyRating difficulty, bool useLighterCo
             }
         }
 
+        public Color4 ForOverlayElement(OverlayColourScheme colourScheme, float saturation, float lightness, float opacity = 1) => Color4.FromHsl(new Vector4(getBaseHue(colourScheme), saturation, lightness, opacity));
+
+        // See https://github.com/ppy/osu-web/blob/4218c288292d7c810b619075471eaea8bbb8f9d8/app/helpers.php#L1463
+        private static float getBaseHue(OverlayColourScheme colourScheme)
+        {
+            float hue;
+
+            switch (colourScheme)
+            {
+                default:
+                    throw new ArgumentException(@"Used colourScheme has no hue value!");
```

Comment:
The exception message doesn't seem to be descriptive enough:
```suggestion
                    throw new ArgumentException($@"{colourScheme} colour scheme does not provide a hue value in {nameof(getBaseHue)}.");
```